TileStache/Goodies/VecTiles/transform.py

```diff
@@ -291,44 +291,45 @@ def road_classifier(shape, properties, fid, zoom):
 def road_sort_key(shape, properties, fid, zoom):
     # Note! parse_layer_as_float must be run before this filter.
 
-    # Calculated sort value is in the range 0 to 39
-    sort_val = 0
+    floor = 300
+    ceiling = 385
+    sort_val = floor
 
-    # Base layer range is 15 to 24
     highway = properties.get('highway', '')
     railway = properties.get('railway', '')
     aeroway = properties.get('aeroway', '')
     aerialway = properties.get('aerialway', '')
     service = properties.get('service')
 
-    is_railway = railway in ('rail', 'tram', 'light_rail', 'narrow_guage', 'monorail')
+    is_railway = railway in (
+        'rail', 'tram', 'light_rail', 'narrow_guage', 'monorail')
 
     if highway == 'motorway':
-        sort_val += 24
+        sort_val += 44
     elif is_railway:
-        sort_val += 23
+        sort_val += 43
     elif highway == 'trunk':
-        sort_val += 22
+        sort_val += 42
     elif highway == 'primary':
-        sort_val += 21
+        sort_val += 41
     elif highway == 'secondary' or aeroway == 'runway':
-        sort_val += 20
+        sort_val += 40
     elif highway == 'tertiary' or aeroway == 'taxiway':
-        sort_val += 19
+        sort_val += 39
     elif highway.endswith('_link'):
-        sort_val += 18
+        sort_val += 38
     elif highway in ('residential', 'unclassified', 'road', 'living_street'):
-        sort_val += 17
+        sort_val += 37
     elif highway in ('unclassified', 'service', 'minor'):
-        sort_val += 16
+        sort_val += 36
     elif aerialway in ('gondola', 'cable_car'):
-        sort_val += 27
+        sort_val += 47
     elif aerialway == 'chair_lift':
-        sort_val += 26
-    elif aerialway != '':
-        sort_val += 25
+        sort_val += 46
+    elif aerialway:
+        sort_val += 45
     else:
-        sort_val += 15
+        sort_val += 25
```

Comment:
I think in this schema, the default would be `35` (not `25`) – else some tunnels will will be **above** default surface "roads"?

If we want to "pad" the ranges to allow arbitrary (client side) orders, then we should offset all up (inclusive of the default), like you're already doing. If we want more padding, we'd +10 them all again.

**Context:** We want to pad values so buildings, water strokes, transit platforms, and other features can lay above tunnels, but below roads. To do that, I think it's better to -40 for tunnels (so it's symmetrical with +40 for bridges).

We also want to give the railway values their own unique sort_val (not collide with road values).

* other road service (alley, driveway) ≠ other railway service
* unclassified, service, minor ≠ railway yard
* residential, unclassified, road, living_street, Unknown ≠ railway spur, siding

See first comment above for recommendation.